backend/capellacollab/projects/toolmodels/modelsources/git/gitlab/exceptions.py

```diff
@@ -42,7 +42,7 @@ async def gitlab_project_not_found_handler(
     return await exception_handlers.http_exception_handler(
         request,
         fastapi.HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
```

Comment:
404 should be better for project "not found" (the name states it already).

Reply:
I gave it a 422, since the resource is not found in GitLab. But 404 will do it for sure.